### PR TITLE
Update CI to support up to Go v1.17

### PR DESCRIPTION
**Pull request by wingyplus**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -12,7 +12,14 @@ jobs:
...
       matrix:
-        go: [ '1.14', '1.13', '1.12', '1.11' ]
+        go:
+        - 1.11
+        - 1.12
+        - 1.13
+        - 1.14
+        - 1.15
+        - 1.16
+        - 1.17
```
Comment (wingyplus) [inline .github/workflows/go.yml:22]: If running on every version is a bit too much. Maybe we can just support only the last 3 (or 4) versions. ^^

**Comment by kkdai** [comment]
@wingyplus Thank you for your PR, how about we just remain following versions? :

- 1.11 (gomodule enable)
- 1.13 (default vendor on)
- 1.16 (ARM support)
- 1.17 (latest version)

What do you think?

**Comment by wingyplus** [comment]
> @wingyplus Thank you for your PR, how about we just remain following versions? :
> 
>     * 1.11 (gomodule enable)
> 
>     * 1.13 (default vendor on)
> 
>     * 1.16 (ARM support)
> 
>     * 1.17 (latest version)
> 
> 
> What do you think?

I think the 2 last versions would be enough, In this case, 1.16 and 1.17. We may add a beta version (such as 1.18-beta.1) to make it doesn't have bugs for future releases.

**Comment by kkdai** [comment]
image

image


We could stay original modification since it only impact 8s (original 60s ---> 68s)

**Review by kkdai — APPROVED** [review]
LGTM

**Pull request merged**